components/post_emoji/index.jsx

```diff
@@ -12,17 +12,20 @@ import {getEmojiMap} from 'selectors/emojis';
 import PostEmoji from './post_emoji.jsx';
 
 function mapStateToProps(state, ownProps) {
+    const emojiName = ownProps.name;
     const emojiMap = getEmojiMap(state);
-    const emoji = emojiMap.get(ownProps.name);
+    const emoji = emojiMap.get(emojiName);
 
     let imageUrl = '';
     let displayTextOnly = false;
-    if (emoji) {
+    if (state.entities.emojis.nonExistentEmoji.has(emojiName)) {
+        displayTextOnly = true;
+    } else if (emoji) {
         imageUrl = getEmojiImageUrl(emoji);
+    } else if (getConfig(state).EnableCustomEmoji !== 'true' || getCurrentUserId(state) === '') {
+        displayTextOnly = false;
```

Comment:
Is this correct? If custom emojis are disabled we should display only text right?

Reply:
Oh sorry for that, it's not correct.  I've changed the logic now. PTAL, thanks